### ISSUE TITLE
Vanara update causes File browser icons are upside-down.

### ISSUE THREAD

**Issue by tsvietOK** [files-community/Files#3256]
**Describe the bug**
The bug caused by 5a053f85ec5654c1e25f1c273b5ae02bf6671278

**To Reproduce**
1. Go to any folder contains files.

**Screenshots**
![image](https://user-images.githubusercontent.com/20501502/105631949-9b90f200-5e59-11eb-94dd-e68724c2b7fe.png)

**Desktop (please complete the following information):**
 - OS Version: [e.g. Windows 10 20H2 19042.746]
 - App version: [v1.1.6]

**Comment by gave92** [comment]
@dahall was this expected? With 3.3.0 we were getting all the images upright (overlays, files and context-menu icons and thumbnails). With 3.3.2 icons and thumbnails appear consistently upside-down :)

**Comment by dahall** [comment]
@gave92 Can you tell me what code is involved so I can see what changed?

**Comment by gave92** [comment]
Thanks this was the code we were using to [get a thumbnail](https://github.com/files-community/Files/blob/main/Files.Launcher/Win32API.cs#L146):

```cs
using var shellItem = new Vanara.Windows.Shell.ShellItem(path);
if (shellItem.IShellItem is Shell32.IShellItemImageFactory fctry)
{
	var flags = Shell32.SIIGBF.SIIGBF_BIGGERSIZEOK;
	if (thumbnailSize < 80) flags |= Shell32.SIIGBF.SIIGBF_ICONONLY;
	var hres = fctry.GetImage(new SIZE(thumbnailSize, thumbnailSize), flags, out var hbitmap);
	if (hres == HRESULT.S_OK)
	{
		using var image = GetBitmapFromHBitmap(hbitmap);
		if (image != null)
		{
			byte[] bitmapData = (byte[])new ImageConverter().ConvertTo(image, typeof(byte[]));
			iconStr = Convert.ToBase64String(bitmapData, 0, bitmapData.Length);
		}
	}
	//Marshal.ReleaseComObject(fctry);
}

private static Bitmap GetBitmapFromHBitmap(HBITMAP hBitmap)
{
	try
	{
		Bitmap bmp = hBitmap.ToBitmap();
		if (Image.GetPixelFormatSize(bmp.PixelFormat) < 32)
		{
			return bmp;
		}
		if (IsAlphaBitmap(bmp, out var bmpData))
		{
			return GetAlphaBitmapFromBitmapData(bmpData);
		}
		return bmp;
	}
	catch
	{
		return null;
	}
}

private static Bitmap GetAlphaBitmapFromBitmapData(BitmapData bmpData)
{
	return new Bitmap(
			bmpData.Width,
			bmpData.Height,
			bmpData.Stride,
			PixelFormat.Format32bppArgb,
			bmpData.Scan0);
}
```

**Comment by gave92** [comment]
Update, I tried with the following: this seems to be returning upsidedown *icons* but upright *thumbnails*:
```cs
using var shellItem = new ShellItem(path);
var shellImages = new ShellItemImages(shellItem);
var flags = ShellItemGetImageOptions.BiggerSizeOk;
if (thumbnailSize < 80) flags |= ShellItemGetImageOptions.IconOnly;
using var image = await shellImages.GetImageAsync(new Size(thumbnailSize, thumbnailSize), flags);
if (image != null)
{
	using var bitmap = image.ToBitmap();
	byte[] bitmapData = (byte[])new ImageConverter().ConvertTo(bitmap, typeof(byte[]));
	iconStr = Convert.ToBase64String(bitmapData, 0, bitmapData.Length);
}
```

**Comment by yaichenbaum** [comment]
This issue should be fixed with #3279.

**Comment by dahall** [comment]
Thanks @yair100. It appears there is a problem in the HBITMAP.ToBitmap extension method. I'll look into it.
